### ISSUE TITLE
Show Moves

### ISSUE THREAD

**Issue by venounan** [vinnymac/PokeNurse#66]
Being able to see the pokemon's moves would be helpful in determining whether or not it should be transferred.

**Comment by vinnymac** [comment]
We added some of this data to the `baseStats.json`, we have plans on adding it to the detail screen.

**Comment by iPaulis** [comment]
Would you add the dps too? also taking into account STAB for the dps damage and an indicator showing if it is the best possible move for each pokemon? that would be absolutely great.

**Comment by vinnymac** [comment]
@iPaulis we currently don't have all that information. If we did we would add it though. It will probably be added as a separate feature. I've seen some sites with data, but I am not sure of the accuracy of third party information. If pogobuf can give us this info it would be best.

**Comment by iPaulis** [comment]
Here is the info: https://docs.google.com/spreadsheets/d/1hcFo7-UGWx1k1u1BHOvDhq8foPeRr7YbX2jLjjJK0Qw/edit#gid=1036598933

I am not sure of the realiability, but their work is impresive.
Also, another user created a simpler table for the moveset effectiveness for each pokemon taking the data from the previous spreadsheet:
https://docs.google.com/spreadsheets/d/1bZWRNBEjYlKyrkzNV6-wQ1RLq6qj_V54oYkvnKyEMsk/edit#gid=1036598933

Showing some percentage info like this for the attacks in the profile of a specific pokemon would be awesome and keep things visualy simple.

**Comment by iPaulis** [comment]
Sorry for the double comment. I am sure you are aware, but just in case, maybe you could share some resources/materials/data with Blossom's Pokemon go Manager project: https://github.com/Blossomforth/BlossomsPokemonGoManager/

Essentialy both of you are doing a similar thing and need the same data and so on for the many features you share.

**Comment by vinnymac** [comment]
Let me rephrase. We currently don't have data in the format we need to efficiently program the feature. The data exists sure, but we need a way to retrieve the information in JSON format or another useful format so that we can easily present it. Curating the information by hand is a daunting task, and we have some other more pressing changes to make to the project. We have a generateBaseStats task, that currently creates some useful information. The `baseStats.json` would be a good place to look if you wanted to add the necessary information to this project.

Perhaps you could do an export to csv, then csv to json and then process it. Just an idea.

Working with Blossom is a good idea, they have a lot of awesome things going on. Thanks for the links.

**Comment by iPaulis** [comment]
I exported to CSV then to JSON, but I am not sure what you mean by proccess it. I had a look at baseStats.json but I don't know how this works and how to implement the data from the files I'm attaching, I don't want to break anything. I wish I could do more, but my knowledge in this area is very limited. I just wanted to put in my 2 cents with some ideas.
I hope these files can be of some help in the future. I understand the are more concerning priorities right now.

[Pokemon GO Species Data and Moves %.zip](https://github.com/duhminick/PokeNurse/files/414128/Pokemon.GO.Species.Data.and.Moves.zip)

In the "Results" file you will find the different combinations of movesets and a percentage of effectiveness for each combination. This percentage does not only take into account DPS for attack, but also HP assuming the pokemon can stay alive for the whole fight (100s), and it makes the same calculations for defense in gym. Then the percentage is calculated from a attack/defense effectiveness ratio.
For more info about these calculations, go to "Extended Results" file, all data is in there. The comments in the google spreadsheet explain how the calculations were made and they seem pretty solid.

**Comment by vinnymac** [comment]
@iPaulis that is really useful, thanks for the data! I am sure it will come in handy, and the calculations too :) They look like they could be processed and included in the baseStats rather easily.

**Comment by iPaulis** [comment]
I found the source of the data and seems very realiable according to reddit users. It also explains the methodology followed in the calculations. Just an amazing work: https://www.reddit.com/r/TheSilphRoad/comments/4vcobt/posthotfix_pokemon_go_full_moveset_rankings/

**Comment by mackhankins** [comment]
![screen shot 2016-08-15 at 2 24 42 pm](https://cloud.githubusercontent.com/assets/2184080/17676476/585a3f60-62f4-11e6-9361-cd54b372a4bc.png)

**Comment by iPaulis** [comment]
Wow, that is absolutely beautiful, great work.

Does the dps count if the attack is the same type of the pokemon? if it already takes that into account in the calculations, then it would be nice to mark it with some colour or something, so we can know it is not just base dps. If it doesn't calculate, it should, because it is bonus damage with a 1.25 multiplier - the move deals 25% more damage.

Also, related to this, is it possible to have a best move mark? a little mark that shows the move is the one that has the highest dps (including bonus) among the possible moves the pokemon could have.

**Comment by Wrexial** [comment]
Nope i forgot to factor in stab, thanks ill make a note next to DPS.

The second bit would be slightly harder to determine, but ill give it a shot. EDIT: this is actually situational. Given that stab and type advantage both give .25 advantage, if lets say an Arcanine uses fire fang on a Relicanth it'll do 1 \* 0.75 \* 0.75 damage even though stats wise fire fang would be the best for Arcanine. I got another thing in mind but may have to be pushed forward to a later date.

**Comment by iPaulis** [comment]
Thank you. You have the attack combinations ordered by effectiveness in the .zip file uploaded above, if that helps at all.

Besides, to make it easier with no maths or anything, I created this table and converted it to JSON with the data of the best moves used in here https://pokemon.gameinfo.io/ so it is completely realiable (updated to 3rd august changes):
[Best Moves.zip](https://github.com/vinnymac/PokeNurse/files/418899/Best.Moves.zip)

It is just the list of all the pokemon, their best quick move and their best charged move, that's all, plain and simple.

**Comment by iPaulis** [comment]
To determine the damage dealt to another pokemon is a whole new story because of type combinations and so on, but that is not what we need in this software. That would be a fighting simulator. So, for now, leave that to the player, and they will decide which pokemon to use for attacking and what is the most convenient thing to do.
However, what we do need to know and this software could help at, is to determine which one of my 2 arcanines is the one who could output more damage based on the movesets they have and their dps. That is why I shared the JSON with the best moves, because those are simply the best moves a given pokemon could have according to dps (STAB is taken into account). 
More complex calculations could be done using other variables too (in the file I shared a couple of days ago), like HP and tankiness of a pokemon based on their stats, and so on, but best dps move is a good starting point, it is easier to understand and it should be simpler to implement.
